torch/onnx/_internal/exporter/_compat.py

```diff
@@ -52,52 +57,47 @@ def _from_dynamic_axes_to_dynamic_shapes(
     if input_names is None:
         input_names = []
 
-    sig = _signature(model)
-    if len(input_names) > len(sig.parameters):
-        raise ValueError(
-            f"Number of input names ({len(input_names)}) should not be greater than "
-            f"the number of model inputs ({len(sig.parameters)})"
-        )
-    input_names_to_model_inputs = {}
-    for idx, param_name in enumerate(sig.parameters):
-        if idx < len(input_names):
-            input_names_to_model_inputs[input_names[idx]] = param_name
-        else:
-            input_names_to_model_inputs[param_name] = param_name
+    if kwargs is None:
+        kwargs = {}
 
-    # NOTE: torch.export.export does not support input names assignment,
-    # so we need to map input names to model inputs to create dynamic_shapes
-    # for the exported program
-    dynamic_shapes_to_exported_program = {}
+    dynamic_shapes: dict[str, Any | None] = {}
     for input_name, axes in dynamic_axes.items():
         if input_name in output_names:
             # User specified an output name as a dynamic axis, so we skip it
             continue
-        # input_name can be either from input_names or from the model inputs
-        if input_name not in input_names_to_model_inputs:
-            raise ValueError(
-                f"dynamic axis: {input_name} is not found in the input names: {input_names}"
-            )
-        model_input_name = input_names_to_model_inputs[input_name]
         if isinstance(axes, dict):
-            dynamic_shapes_to_exported_program[model_input_name] = {
-                k: torch.export.Dim(v) for k, v in axes.items()
+            # Dim needs to pass str.isidentifier()
+            dynamic_shapes[input_name] = {
+                k: torch.export.Dim(re.sub(r"[^A-Za-z_]", "", v))
```

Comment:
Is this removing all the illegal characters?

Reply:
Sorry I misread. Could you say more on what this does?

Reply:
This helps to avoid hitting https://github.com/pytorch/pytorch/blob/c98ef0279e6eb968f5f9d22e1f193e7064594152/torch/export/dynamic_shapes.py#L230

Comparing to pure string that dynamic_axes uses, torch.export.Dim has more restrictions. This re.sub only allows alphabet and underscore. But erroring out here also makes sense, as users might not want us to modify their naming, or names could be duplicated after modified.